frontend/src/components/ViewTrips/index.js

```diff
@@ -36,36 +45,84 @@ class ViewTrips extends React.Component {
   }
 
   /**
-   * Flips `refreshAddTripModal` property which causes that AddTripModal
-   * component to be reloaded.
+   * Handler that flips `refreshSaveTripModal` property which causes that
+   * save/edit trip component to be reloaded.
    *
    * This was done to prevent bugs where multiple component input fields would
    * persist from one instance of the modal to the next when view trips page
    * was not refreshed in between.
    */
-  refreshAddTripModal = () => {
-    this.setState({ refreshAddTripModal: !this.state.refreshAddTripModal });
+  refreshSaveTripModal = () => {
+    this.setState({ refreshSaveTripModal: !this.state.refreshSaveTripModal });
   }
 
-  /** Property that refreshes and displays add trip page. */
-  showAddTripModal = () => {
-    this.refreshAddTripModal()
+  /** Handler that hides the add/edit trip page. */
+  hideSaveTripModal = () => { this.setState({ showModal: false }); }
+
+  /** Handler that refreshes and displays the add/edit trip page. */
+  showSaveTripModal = () => {
+    this.refreshSaveTripModal();
     this.setState({ showModal: true });
   }
 
-  /** Property that sets `showModal` to false --> hides add trip page. */
-  hideAddTripModal = () => { this.setState({ showModal: false }); }
+  /**
+   * Handler that displays the add trip page.
+   *
+   * Sets state for the states `tripId` and `placeholderObj` in order
+   * to ensure the modal has the visual characteristics of an "add trip" modal
+   * and creates a new Trip document in the database.
+   */
+  showAddTripModal = () => {
+    this.setState({
+      tripId: null,
+      placeholderObj: {
+        name:          'Enter Trip Name',
+        description:   'Enter Trip Description',
+        destination:   'Enter Trip Destination',
+        startDate:     '',
+        endDate:       '',
+        collaborators: ['person@email.xyz']
```

Comment:
No it is supposed to hold the placeholder for now (this is changed in #73). The 'infrastructure' is slightly changed in #73 but this idea was that this value would be seen in the email field when creating a new trip. And when editing an existing trip, the existing collaborator email will show up rather than the placeholder.